Automattic/jetpack
# Update noticon from notification to bell

Fixes https://github.com/Automattic/jetpack/issues/36192

This updates the notification icon used in the top bar when the Masterbar is not being rendered.

## Changes

The only change here is to the notification HTML markup, its slightly different. 
It used to be;

```
<a class="ab-item" href="https://wordpress.com/notifications">
	<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read">
		<span class="screen-reader-text">Notifications</span>
		<span class="noticon noticon-notification"></span>
	</span>
</a>
```

And now it will be the same as on Masterbar;

```
<a class="ab-item" href="https://wordpress.com/notifications">
	<span id="wpnt-notes-unread-count" class="wpnt-loading wpn-read"></span>
	<span class="noticon noticon-bell"></span>
	<span class="screen-reader-text">Notifications</span>
</a>
```

This PR adds a new static function  `get_notes_markup` to return the markup used.

The CSS changes are done to the following CSS file;

`https://s0.wp.com/wp-content/mu-plugins/notes/admin-bar-v2.css`

Those changes are done on WPCOM sandbox on D141499-code.

We need to add a top level class to the `#wp-admin-bar-notes` li element so that we can support styling the bell icon while also supporting the exisiting notification icon on older verisons of Jetpack.

## Before

https://github.com/Automattic/jetpack/assets/5560595/8fc08e00-4346-45f8-a4ca-23fb20c06e48

## After

https://github.com/Automattic/jetpack/assets/5560595/7a34521b-e8ab-4516-97ac-defba0636659

## Testing instructions:

* Apply CSS changes in D141499-code
* Add `define( 'JETPACK_NOTES__CACHE_BUSTER', time() );` to your 0-sandbox.php file
* Follow setup guidelines here - pdWQjU-77-p2
* Create JN site
* Complete Jetpack setup and select free plan
* Confirm notification looks/behaves the same as notification icon on simple site
* Create JN site but don't sync PR, confirm the older notification icon looks/behaves ok (pictures below)
<img width="227" alt="Screenshot 2024-03-14 at 12 25 37" src="https://github.com/Automattic/jetpack/assets/5560595/1df7a3d3-7780-4e45-81d3-d0fc5f7cb156">
<img width="187" alt="Screenshot 2024-03-14 at 12 25 23" src="https://github.com/Automattic/jetpack/assets/5560595/713a7b0c-1c1d-4438-a143-d12706b6053f">

## Does this pull request change what data or activity we track or use?

No
